service/src/main/java/org/databiosphere/workspacedataservice/jobexec/JobDataMapReader.java

```diff
@@ -84,4 +85,17 @@ public boolean getBoolean(String key) {
           e);
     }
   }
+
+  /**
+   * Retrieve a value from a JobDataMap. Throws a JobExecutionException if the value is not
+   * found/null.
+   */
+  public Object get(String key) {
+    try {
+      return Objects.requireNonNull(jobDataMap.get(key));
+    } catch (Exception e) {
+      throw new JobExecutionException(
+          "Error retrieving key %s from JobDataMap: %s".formatted(key, e.getMessage()), e);
+    }
+  }
```

Comment:
By adding a type parameter here, you could add validation and prevent call sites from having to do their own unchecked casts. Any `ClassCastException` would get caught and rethrown as `JobExecutionException`:
```suggestion
  /**
   * Retrieve a value from a JobDataMap. Throws a JobExecutionException if the value is not
   * found/null or of the wrong type.
   */
  @SuppressWarnings("unchecked")
  public <T> T get(String key) {
    try {
      return (T) Objects.requireNonNull(jobDataMap.get(key));
    } catch (Exception e) {
      throw new JobExecutionException(
          "Error retrieving key %s from JobDataMap: %s".formatted(key, e.getMessage()), e);
    }
  }
```

Reply:
👍 I tried that at first, but Intellij complained about an unchecked cast, which it didn't in ImportDetailsRetriever.

Reply:
`@SuppressWarnings("unchecked")` ftw :)